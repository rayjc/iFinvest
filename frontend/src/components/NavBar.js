import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';


const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  logoutButton: {
    marginLeft: theme.spacing(2),
  }
}));

const NavBar = ({ setShowDrawer, handleLogout }) => {
  const classes = useStyles();
  const user = useSelector(state => state.user.user);

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton edge="start" className={classes.menuButton}
            color="inherit" aria-label="menu" onClick={() => setShowDrawer(true)}>
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            iFinvest
          </Typography>
          {user ?
            <>
              <IconButton
                aria-label="account of current user"
                color="inherit"
                component={Link}
                to="/profile"
              >
                <AccountCircle />
              </IconButton>
              <IconButton
                className={classes.logoutButton}
                aria-label="log out"
                color="inherit"
                onClick={handleLogout}
              >
                <ExitToAppIcon />
              </IconButton>
            </> :
            <Button color="inherit" component={Link} to="/auth">Login</Button>
          }
        </Toolbar>
      </AppBar>
    </div>
  );
};


export default NavBar;