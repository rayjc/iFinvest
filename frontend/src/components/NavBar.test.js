import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { fireEvent, render } from '@testing-library/react';
import NavBar from './NavBar';
import { useSelector, useDispatch } from 'react-redux';


const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch
}));

// smoke test
it("renders without crashing", function() {
  render(<MemoryRouter><NavBar /></MemoryRouter>);
});

// snapshot test
it("matches snapshot", function() {
  const { asFragment } = render(<MemoryRouter><NavBar /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});


describe("NavBar buttons for annoymous user", function() {
  it("has login button", function() {
    const { getByText } = render(<MemoryRouter><NavBar /></MemoryRouter>);
    expect(getByText("Login")).toBeInTheDocument();
  });
});

describe("NavBar buttons for logged in user", function() {
  beforeEach(function() {
    useSelector.mockImplementation((selectorFn) =>
      selectorFn({
        user: {
          user: {
            id: 0, username: "test", first_name: "test", last_name: "user",
            email: "test@test.com"
          },
          error: null, isFetching: false,
        }
      })
    );
  });

  it("has profile button", function() {
    const { getByLabelText } = render(<MemoryRouter><NavBar /></MemoryRouter>);
    expect(getByLabelText("account of current user")).toBeInTheDocument();
  });

  it("has logout button", function() {
    const { getByLabelText } = render(<MemoryRouter><NavBar /></MemoryRouter>);
    expect(getByLabelText("log out")).toBeInTheDocument();
  });
});