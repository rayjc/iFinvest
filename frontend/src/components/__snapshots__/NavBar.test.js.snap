// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-5"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-7 MuiTypography-h6"
        >
          iFinvest
        </h6>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/auth"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
